clickhouse_search/backend/fields.py

```diff
@@ -15,13 +19,26 @@ def db_type(self, connection):
     def cast_db_type(self, connection):
         return super().cast_db_type(connection).replace('Tuple', 'Nested', 1)
 
+    def from_db_value(self, *args, **kwargs):
+        return self._from_db_value(*args, **kwargs)
+
     def _from_db_value(self, value, expression, connection):
         if value is None:
             return value
-        return [self.container_class(*item) for item in value]
+        value = [self._convert_type(item)._asdict() for item in value]
+        if self.group_key:
+            value = {item[self.group_key]: item for item in value}
+        return value
 
 
 class UInt64FieldDeltaCodecField(models.UInt64Field):
 
     def db_type(self, connection):
         return f'{super().db_type(connection)} CODEC(Delta(8), ZSTD(1))'
+
+
+class NamedTupleField(models.TupleField):
+
+    def _convert_type(self, value):
+        value = super()._convert_type(value)
+        return value._asdict() if value is not None else value
```

Comment:
Django is correctly returning named Tuple fields as python named tuples, however there is a bug with python's `json.dumps` where named tuples are serialized as lists (and the names are dropped). Therefore, I overrode the behavior for serializing the field here so it returns a dictionary instead